### PR TITLE
Allow donation address to be updated

### PR DESCRIPTION
**Pull request by Abban**
Also removes deprecated newsletter and receipt opt-ins.

Ticket (address update): https://phabricator.wikimedia.org/T316632
Ticket (is exported): https://phabricator.wikimedia.org/T326246

### CODE REVIEW

**Review by moiikana — APPROVED** [review]

**Review by gbirke — CHANGES_REQUESTED** [review]
We are striving to remove the test templates (a leftover from the old PHP-webapp-centered days), see https://phabricator.wikimedia.org/T163452

Please take the opportunity to change the tests and delete the template files.

**Comment by moiikana** [comment]
there is also a test missing here that needs to be fixed, related to that:
UpdateDonorRouteTest::testWhenDonationAlreadyHasAddress_requestIsDenied
the assertion needs to assume false

**Comment by Abban** [comment]
> there is also a test missing here that needs to be fixed

Huh, I deleted that already and it came back.

**Comment by Abban** [comment]
Oh I see, it is deleted. We no longer need that test though, as we don't have that functionality there's no branch to test anymore.

**Review by gbirke — APPROVED** [review]

File: skins/test/templates/Donation_Confirmation.html.twig [hunk 1]
```diff
@@ -22,7 +22,7 @@
 	</p>
 {% endif %}
 
-{% if donation.optsIntoNewsletter == true %}
+{% if donation.newsletter == true %}
```
Comment (gbirke) [inline skins/test/templates/Donation_Confirmation.html.twig:25]: You can delete this file now as well

**Pull request merged**